src/Desktop/UIAutomation/CustomObjects/Registrar.cs

```diff
@@ -9,21 +14,59 @@ namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
 {
     public class Registrar : IDisposable
     {
-        CUIAutomationRegistrar _uiaRegistrar = new CUIAutomationRegistrar();
+        private IUIAutomationRegistrar _uiaRegistrar;
+        private Action<int, ITypeConverter> _callback;
         private bool disposedValue;
 
-        public int RegisterCustomProperty(Guid id, string name, UIAutomationType type)
+        public Registrar()
+            : this(new CUIAutomationRegistrar(), new Action<int, ITypeConverter>(A11yProperty.RegisterCustomProperty)) { }
+
+        public void RegisterCustomProperty(CustomProperty prop)
         {
+            if (prop == null) throw new ArgumentNullException(nameof(prop));
             UIAutomationPropertyInfo info = new UIAutomationPropertyInfo
             {
-                guid = id,
-                pProgrammaticName = name,
-                type = type,
+                guid = prop.Guid,
+                pProgrammaticName = prop.ProgrammaticName,
+                type = GetUnderlyingUIAType(prop.Type)
             };
 
             _uiaRegistrar.RegisterProperty(ref info, out int dynamicId);
+            _callback(dynamicId, CreateTypeConverter(prop.Type));
+        }
 
-            return dynamicId;
+        internal Registrar(IUIAutomationRegistrar uiaRegistrar, Action<int, ITypeConverter> callback)
+        {
+            _uiaRegistrar = uiaRegistrar;
+            _callback = callback;
+        }
+
+        private static UIAutomationType GetUnderlyingUIAType(CustomUIAPropertyType type)
+        {
+            switch (type)
+            {
+                case CustomUIAPropertyType.String: return UIAutomationType.UIAutomationType_OutString;
+                case CustomUIAPropertyType.Int: return UIAutomationType.UIAutomationType_Int;
+                case CustomUIAPropertyType.Bool: return UIAutomationType.UIAutomationType_Bool;
+                case CustomUIAPropertyType.Double: return UIAutomationType.UIAutomationType_Double;
+                case CustomUIAPropertyType.Point: return UIAutomationType.UIAutomationType_OutPoint;
+                case CustomUIAPropertyType.Element: return UIAutomationType.UIAutomationType_Element;
+                default: throw new ArgumentException("Unset or unknown type", nameof(type));
```

Comment:
Do we have a unit test that triggers this Exception?

Reply:
Yes, `CreateTypeConverterUnsetTest`.